### DISCUSSION TITLE
Working local.yaml configuration for Azure SAML

### DISCUSSION THREAD

**Discussion by irfanahmed786** [kimai/kimai#4964] [Docker]
Can someone share the local.yaml file for Azure SAML? I am unable to configure Azure SAML on my on prem environment.
Do any additional steps need to be followed, which I might miss?
I followed documentation, which is there on the Kimai website.

**Comment by irfanahmed786** [comment]
Any help would be appreciated

**Comment by josh-go-tech** [comment]
```kimai:
    saml:
        provider: microsoft
        activate: true
        title: Login with Entra ID
        mapping:
            - { saml: $http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name, kimai: username }
            - { saml: $http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress, kimai: email }
            - { saml: $http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname $http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname, kimai: alias }
        roles:
            resetOnLogin: true
            attribute: http://schemas.microsoft.com/ws/2008/06/identity/claims/groups
            mapping:
                - { saml: xxxxxxxx-yyyy-xxxx-yyyy-xxxxxxxxxxxx, kimai: ROLE_ADMIN }
                - { saml: Azure-Group-Object-Id, kimai: ROLE_TEAMLEAD }
        connection:
            idp:
                entityId: 'https://sts.windows.net/******-*-****-****-****-***********/'
                singleSignOnService:
                    url: 'https://login.microsoftonline.com/******-*-****-****-****-***********/saml2'
                x509cert: 'CERT GOES HERE'
            sp:
                entityId: 'https://mydomain.com/auth/saml/metadata'
                assertionConsumerService:
                    url: 'https://mydomain.com/auth/saml/acs'
                singleLogoutService:
                    url: 'https://mydomain.com/auth/saml/logout'
            debug: true
            security:
                requestedAuthnContext: false

**Reply by irfanahmed786** [reply]
How to put certificate in x509cert? I am getting Authenticator error while login with SAML.

**Reply by irfanahmed786** [reply]
I can see warning message in log file saying "x.509 certificate cannot be retrieved"

**Comment by irfanahmed786** [comment]
@kevinpapst, can you please check and help on this? I want to configure SSO but encountering with this issue.

**Reply by kevinpapst** [reply]
It's all in the docs and works as expected.
If it fails for you, your config is invalid.

**Reply by irfanahmed786** [reply]
Thank you !! @kevinpapst for your response.
Actually, I followed the document and updated the information accordingly. Can you give me some sample information what I can put under x509cert:

**Reply by kevinpapst** [reply]
If you have a cert like that:
![image](https://github.com/user-attachments/assets/537dfe80-a01c-4601-9bcb-f581215d03ff)

You remove the start and end line and then merge the entire file into one line (remove all whitespaces):
<img width="447" alt="Bildschirmfoto 2024-08-05 um 09 44 27" src="https://github.com/user-attachments/assets/abc970f2-1231-4e3a-ba6c-6ccf11235f68">

**Reply by irfanahmed786** [reply]
Thank you !! @kevinpapst , Much appreciate your feedback.

I can see the configuration is correct and as per documentations but still can see below warning.

Warning: openssl_x509_export(): X.509 Certificate cannot be retrieved at /var/www/kimai/vendor/robrichards/xmlseclibs/src/XMLSecurityKey.php:366)"} []

**Reply by kevinpapst** [reply]
Share your config if you want help and stop tagging me for no reason

**Comment by wmeneses** [comment]


**Reply by kevinpapst** [reply]
Yes, otherwise it would work...

**Reply by kevinpapst** [reply]
And please stop hijacking issues. Your post does not help solve the original question.
Post a new discussion instead with all required details.
